python-avd/pyavd/_eos_designs/structured_config/network_services/vlans.py

```diff
@@ -98,6 +98,18 @@ def _get_vlan_config(
             name=vlan.name,
             tenant=tenant.name,
         )
+        if vlan.address_locking.ipv4:
+            if self.inputs.address_locking_settings.dhcp_servers_ipv4 and self.inputs.address_locking_settings.local_interface:
+                vlans_vlan.address_locking.address_family.ipv4 = vlan.address_locking.ipv4
+            else:
```

Comment:
There is a use case with IP Locking to prevent rogue DHCP servers using `enforcement disabled` mode that doesn't require DHCP servers or a local interface to be configured. You may want to include that exception in the check here.

Reply:
@jrecchia1029 Can you share TOI for the same, it would get easier for me to implement.

Reply:
Updated description!

Reply:
Looks good to me for this point - thanks @Vibhu-gslab